### PR TITLE
feat: added python unit tests

### PR DESCRIPTION
**Pull request by ziafazal**
PR has 
1. units tests
2. Ability to reset `completed_challenges` challenges when user restarts a track
3. Replaced deprecated `xblock.fragment` with `web_fragments.fragment`

### CODE REVIEW

File: requirements/test.in [hunk 1]
```diff
@@ -5,3 +5,4 @@
 
 pytest-cov                # pytest extension for code coverage statistics
 pytest-django             # pytest extension for better Django support
+xblock-sdk
```
Comment (taimoor-ahmed-1) [inline requirements/test.in]: add blank line

Comment (madnihamza1841) [reply requirements/test.in]: End of File Blank line missing in all files. Please add them as well.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+def instruqt_xblock(request):
+    """Instruct XBlock pytest fixture."""
+    runtime = WorkbenchRuntime()
+    key_store = DictKeyValueStore()
+    db_model = KvsFieldData(key_store)
+    ids = generate_scope_ids(runtime, 'instruqtxblock')
+    instruqt_xblock = InstruqtXBlock(runtime, db_model, scope_ids=ids)
+    instruqt_xblock.usage_id = Mock()
+    instruqt_xblock.has_score = True
+    request.cls.instruqt_xblock = instruqt_xblock
```
Comment (madnihamza1841) [inline tests/conftest.py:32]: Add a newline at the end of file in all files.

**Review by taimoor-ahmed-1 — APPROVED** [review]
LGTM 👍

**Pull request merged**